src/Analysis/Ast/Impl/Analyzer/ModuleWalker.cs

```diff
@@ -83,9 +97,18 @@ public ModuleWalker(IServiceContainer services, IPythonModule module, PythonAst
             SymbolTable.EvaluateAll();
             SymbolTable.ReplacedByStubs.Clear();
             MergeStub();
+
+            if (_allReferencesCount == 1 && GlobalScope.Variables.TryGetVariable(AllVariableName, out var variable) && variable?.Value is IPythonCollection collection) {
+                ExportedMemberNames = collection.Contents
```

Comment:
I'm not sure I really like the idea of exposing a public property which may or may not contain the true list of exports from a module. I think keeping this as an implementation detail for `GetMemberNames` makes more sense, especially since I'm going to be handling more things and better implementing the list of exports (and now that IC is back linking to the analyzer assembly).

There are three scenarios:

1. `__all__` was never assigned.
2. `__all__` was assigned in a way we can handle.
3. `__all__` was assigned in a way we can't handle.
4. `__all__` is defined by `__all__ = []` (a subset of 2).

With this PR, case 1 means `ExportedMemberNames` is empty; case 2 means `ExportedMemberNames` is not empty and `__all__` is an assigned variable in the scope; case 3 means `ExportedMemberNames` is empty but `__all__` is still potentially assigned; case 4 has an empty `ExportedMemberNames` and `__all__` is defined and has a value.

This feels a little inconsistent, whereas previously there was a single source of truth, `GetMemberNames`. Is there something `ExportedMemberNames` is doing that `GetMemberNames` was not, other than that it implies `_allReferencesCount == ` earlier than `GetMemberNames` would determine?

Reply:
I'm not sure I understand the question. `IDocumentAnalysis` doesn't have `GetMemberNames` - it has `GlobalScope`, and `PythonModule.GetMemberNames()` is still a single source of truth.

We can make implementation of `PythonModule.GetMemberNames()` look like:
```cs
public virtual IEnumerable<string> GetMemberNames() 
    => Analysis.ExportedMemberNames;
```
and assign variables from GlobalScope to it in `ModuleWalker.Complete`, something like:
```cs
if (_allReferencesCount == 1 && GlobalScope.Variables.TryGetVariable(AllVariableName, out var variable) && variable?.Value is IPythonCollection collection) {
    ExportedMemberNames = collection.Contents
        .OfType<IPythonConstant>()
        .Select(c => c.GetString())
        .Where(s => !string.IsNullOrEmpty(s))
        .ToImmutableArray();
} else {
    ExportedMemberNames = GlobalScope.Variables
        .Where(v => !(v.Value?.GetPythonType() is PythonModule)
                    && !(v.Value?.GetPythonType().DeclaringModule is TypingModule && !(this is TypingModule)))
        .Select(v => v.Name)
        .ToImmutableArray();
}
```
I didn't think much about it cause there is #620 and current implementation is temporary, but we definitely can make `ExportedMemberNames` a list of exported variables for all cases.

Reply:
Yeah, my point in comments was that #620 is primary and this implementation is temporary and basic.